codegen/src/test/resources/software/amazon/awssdk/codegen/poet/waiters/query-async-waiter-interface.java

```diff
@@ -35,11 +37,23 @@
      *
      * @return a builder
      */
-    static void builder() {
-        throw new UnsupportedOperationException();
+    static Builder builder() {
+        return DefaultQueryAsyncWaiter.builder();
     }
 
     interface Builder {
+        /**
+         * Sets a custom {@link ScheduledExecutorService} that will be used to schedule async polling attempts
+         * <p>
+         * This executorService must be closed by the caller when it is ready to be disposed. The SDK will not close the
+         * executorService when the waiter is closed
+         *
+         * @param executorService
+         *        the executorService to set
+         * @return a reference to this object so that method calls can be chained together.
+         */
+        Builder executorService(ScheduledExecutorService executorService);
```

Comment:
Should we have provider overload, where the returned executors are managed by the SDK?

Reply:
I like the idea, but we don't have it for the futureCompletionExecutorService override either. I feel this is additive and we can add it in the future if people are asking for it. I'd expect most customers use the waiters retrieved the service clients, where they don't need to configure it at all.

Reply:
Sounds good